File: app/src/main/java/com/boardgamegeek/provider/PlayerColorsProvider.java
package com.boardgamegeek.provider;

import android.net.Uri;

import com.boardgamegeek.provider.BggContract.PlayerColors;
import com.boardgamegeek.provider.BggDatabase.Tables;

public class PlayerColorsProvider extends BasicProvider {
    @Override
    protected String getDefaultSortOrder() {
        return PlayerColors.DEFAULT_SORT;
    }

    @Override
    protected String getPath() {
        return BggContract.PATH_PLAYER_COLORS;
    }

    @Override
    public String getTable() {
        return Tables.PLAYER_COLORS;
    }

    @Override
    protected String getType(Uri uri) {
        return PlayerColors.CONTENT_TYPE;
    }
}

File: app/src/main/java/com/boardgamegeek/provider/PlayerColorsProvider.kt
package com.boardgamegeek.provider

import android.net.Uri
import com.boardgamegeek.provider.BggContract.PATH_PLAYER_COLORS
import com.boardgamegeek.provider.BggContract.PlayerColors
import com.boardgamegeek.provider.BggDatabase.Tables

class PlayerColorsProvider : BasicProvider() {
    override fun getType(uri: Uri) = PlayerColors.CONTENT_TYPE

    override val path = PATH_PLAYER_COLORS

    override val table = Tables.PLAYER_COLORS

    override val defaultSortOrder = PlayerColors.DEFAULT_SORT
}